x/auth/keeper/keeper.go

```diff
@@ -275,3 +278,48 @@ func (ak AccountKeeper) GetParams(ctx context.Context) (params types.Params) {
 	}
 	return params
 }
+
+func (ak AccountKeeper) NonAtomicMsgsExec(ctx context.Context, signer sdk.AccAddress, msgs []sdk.Msg) ([]*types.NonAtomicExecResult, error) {
+	var msgsAny []*codectypes.Any
+	msgResponses := make([]*types.NonAtomicExecResult, 0, len(msgs))
+
+	for _, msg := range msgs {
+		if m, ok := msg.(sdk.HasValidateBasic); ok {
+			if err := m.ValidateBasic(); err != nil {
+				value := &types.NonAtomicExecResult{Error: err.Error()}
+				msgResponses = append(msgResponses, value)
+				continue
+			}
+		}
+
+		anyValue, err := codectypes.NewAnyWithValue(msg)
+		if err != nil {
+			value := &types.NonAtomicExecResult{Error: err.Error()}
+			msgResponses = append(msgResponses, value)
+			continue
+		}
+		msgsAny = append(msgsAny, anyValue)
+
+	}
+
+	if err := ak.environment.BranchService.Execute(ctx, func(ctx context.Context) error {
+		results, err := ak.AccountsModKeeper.SendAnyMessages(ctx, signer, msgsAny)
```

Comment:
how come we need to import accounts to use this one function? it seems like we can go without it and possibly duplicate it for a short time.

Reply:
we could do without accounts too, but frojdi mentioned that we could directly use accounts `SendAnyMessages` to execute the msgsAny instead of even needing to unpack them and checking the signers for msgs explicitly